searxng/searxng
# [mod] client_settings: pass settings from server to JS client

## What does this PR do?

[mod] client_settings: pass settings from server to JS client 

- The first commit implements a convenient method to pass settings from the server to the JS client.
- The second commit demonstrates how easy it is to add a setting on server side and use it in the client (JS).

The second commit is related to https://github.com/searxng/searxng/discussions/1082 and an alternative of https://github.com/searxng/searxng/pull/1087

## Why is this change important?

In current master branch it is hard to add a setting and pass it to the client (JS).  ATM we share some settings and properties from server to client, but the implementation for each value is spread over the HTML templates, the render function and needs to be fetched on the JS side.  Without going into details, here is e.g. what is needed on the JS side:

https://github.com/searxng/searxng/blob/09230bfad9dea35a6ed112ea7096935d1c821bd9/searx/static/themes/simple/src/js/head/00_init.js#L17-L29

IMO that is to complicated if we need to share additional settings with the client side .. and https://github.com/searxng/searxng/pull/ shows me, that there is a need for.

## Author's checklist

To spare a additional request / response cycle (_for lees the 1kByte data_) the settings are shipped to the client in a base64 encoded JSON string in an attribute of the `base.html` template.

> the base64 encoding creates a magic value in the HTML page, it's not good for the transparency.

The alternative is to add an additional `<script src="client_settings.js">` to the `base.html` template in which the settings are transferred to client side .. but this will add an addition request / response cycle on each page load (take into mind: `client_settings.js` is not a static file / can't be cached on client side). 

## How to test this PR locally?

Use `make run`, activate autocomplete and change the value of `autocomplete_min` .. observe how autocomplete works in your client. 

## Related issues

- https://github.com/searxng/searxng/discussions/1082
- alternative of https://github.com/searxng/searxng/pull/1087
